### PR TITLE
expression: fix return type of agg func `bit_or` when handling varbinary column

### LINKED ISSUE #36400: Unexpected `bit_or` agg result on varbinary column
## Bug Report

Please answer these questions before submitting your issue. Thanks!

may caused by #36258

### 1. Minimal reproduce step (Required)

```
drop table if exists t;
create table t (a timestamp, b varbinary(32));
insert into t values ('2006-08-27 21:57:57', 0x373037343631313230);
select bit_or(a * b) from t;
```

### 2. What did you expect to see? (Required)

the results of using unistore and tikv are same.

### 3. What did you see instead (Required)

use unistore

```mysql> select bit_or(a * b) from t;
ERROR 1690 (22003): constant 1.4192255290185928e+22 overflows bigint
```

use tikv

```
mysql> select bit_or(a * b) from t;
+---------------------+
| bit_or(a * b)       |
+---------------------+
| 9223372036854775807 |
+---------------------+
1 row in set, 1 warning (0.00 sec)

```

### 4. What is your TiDB version? (Required)

master

### PR DESCRIPTION
**Pull request by solotzg**
### What problem does this PR solve?

Issue Number: close #36400

Problem Summary:

- to fix https://github.com/tikv/copr-test/pull/159
- verified in https://ci.pingcap.net/blue/organizations/jenkins/tidb_ghpr_integration_copr_test/detail/tidb_ghpr_integration_copr_test/9782/pipeline/88

### What is changed and how it works?

- make type cast to int explicit when using `bit_or`, `bit_and`, `bit_xor`

### Check List

Tests 

- [x] Unit test
- [x] Integration test
- [ ] Manual test (add detailed scripts or steps below)
- [ ] No code

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note

Please refer to [Release Notes Language Style Guide](https://pingcap.github.io/tidb-dev-guide/contribute-to-tidb/release-notes-style-guide.html) to write a quality release note.

```release-note
None
```

### CODE REVIEW

**Comment by solotzg** [comment]
@zanmato1984 @XuHuaiyu @guo-shaoge PTAL

File: expression/aggregation/base_func.go [hunk 1]
```diff
@@ -104,6 +104,8 @@ func (a *baseFuncDesc) TypeInfer(ctx sessionctx.Context) error {
 		a.typeInfer4MaxMin(ctx)
 	case ast.AggFuncBitAnd, ast.AggFuncBitOr, ast.AggFuncBitXor:
 		a.typeInfer4BitFuncs(ctx)
+		// make cast to int explicit
+		a.Args[0] = expression.WrapWithCastAsInt(ctx, a.Args[0])
```
Comment (zanmato1984) [inline expression/aggregation/base_func.go:108]: Can we make this implicit cast into function `typeInfer4BitFuncs`? Like https://github.com/solotzg/tidb/blob/eeb23695b8fb913c4a3d92f3528ed9b7f43f537d/expression/aggregation/base_func.go#L273 does?

File: expression/aggregation/base_func.go [hunk 2]
```diff
@@ -287,7 +287,8 @@ func (a *baseFuncDesc) typeInfer4BitFuncs(ctx sessionctx.Context) {
 	a.RetTp.SetFlen(21)
 	types.SetBinChsClnFlag(a.RetTp)
 	a.RetTp.AddFlag(mysql.UnsignedFlag | mysql.NotNullFlag)
-	// TODO: a.Args[0] = expression.WrapWithCastAsInt(ctx, a.Args[0])
+	// make cast to int explicit
```
Comment (zanmato1984) [inline expression/aggregation/base_func.go:290]: Normally we call this an "implicit cast". "Explicit" doesn't fit this case.

Comment (zanmato1984) [reply expression/aggregation/base_func.go:290]: Maybe we should just remove this comment as the code below pretty much tells for itself.

**Review by zanmato1984 — APPROVED** [review]
LGTM

**Review by guo-shaoge — APPROVED** [review]

**Pull request merged**